cdi-core/src/main/java/com/linkedin/cdi/connection/HttpConnection.java

```diff
@@ -96,7 +96,7 @@ synchronized HttpClient getHttpClient(State state) {
         ConnectionClientFactory factory = (ConnectionClientFactory) factoryClass.newInstance();
         httpClient = factory.getHttpClient(state);
       } catch (Exception e) {
-        LOG.error("Error creating HttpClient: {}", e.getMessage());
+        LOG.error("Error creating HttpClient: {}", e.toString());
```

Comment:
Don't do `toString()` here and other locations as well.

Reply:
@toString() prints the name and description of the exception compared to getMessge() which only prints the description of the message which gives more information which can be useful is some scenarios.

@booddu  Please let me know if you if you think otherwise

Reply:
Check out this api `public void error(String msg, Throwable t)`
It will print the whole stack trace. 

You usually don't do toString with log.error